src/index.ts

```diff
@@ -15,26 +15,25 @@
  */
 
 import * as extend from 'extend';
+import * as gcpMetadata from 'gcp-metadata';
 import * as path from 'path';
-const gcpMetadata = require('gcp-metadata');
+import * as pify from 'pify';
 
 import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
 
-import {Config, defaultConfig, internalConfig} from './config';
+import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
-const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
 
 // Returns value of metadata field.
 // Throws error if there is a problem accessing metadata API.
 function getField(field: string): Promise<string> {
-  return new Promise<string>((resolve, reject) => {
-    gcpMetadata.instance(
-        field, (err: Error, response: any, metadata: string) => {
-          err ? reject(err) : resolve(metadata);
-        });
-  });
+  return pify(gcpMetadata.instance, {multiArgs: true})(field).then(
```

Comment:
I think you can store the pify return value at the top-level.

```
const metadataInstance = pify(gcpMetadata.instance, {multiArgs: true});
```
I wonder if there would be much left in this function after that, but I'll let you decide.

Reply:
If you decide to keep this function, use a different make, e.g. `getMetadataInstanceField`.